### PR TITLE
Update postgres install centos

### PR DESCRIPTION
**Pull request by sandoche2k**

### CODE REVIEW

File: docs/Installation.md [hunk 1]
```diff
@@ -118,19 +118,51 @@ sudo install --mode=755 ./backend_config.ini-postgresql /etc/zonemaster/backend_
...
+```sh
+sudo yum -y install postgresql93 postgresql93-server postgresql93-contrib postgresql93-libs postgresql93-devel
+sudo cpan -i DBD::Pg
+```
+
+Configure:
 
 ```sh
+# Switch to root user
+sudo -s
```
Comment (matsduf) [inline docs/Installation.md]: Why do you switch to root when you use sudo in the next command?

File: docs/Installation.md [hunk 2]
```diff
@@ -118,19 +118,51 @@ sudo install --mode=755 ./backend_config.ini-postgresql /etc/zonemaster/backend_
...
+sudo /usr/pgsql-9.3/bin/postgresql93-setup initdb
+
+# Start PostgreSQL
+sudo systemctl start postgresql-9.3 
+
+# Verify PostgreSQL has started
+sudo systemctl status postgresql-9.3
+
+# Add Zonemaster user
+sudo useradd zonemaster
```
Comment (matsduf) [inline docs/Installation.md]: This step is not related to PostgreSQL. I suggest that it is moved to the block where the backend_config.ini is copied. It is also needed for MySQL.

File: docs/Installation.md [hunk 3]
```diff
@@ -118,19 +118,51 @@ sudo install --mode=755 ./backend_config.ini-postgresql /etc/zonemaster/backend_
 sudo install --mode=755 ./zm-centos.sh-postgresql /etc/init.d/zm-centos.sh
 mkdir "$HOME/logs"
 ```
+Get the appropriate PostgreSQL version
```
Comment (matsduf) [inline docs/Installation.md]: I suggest "Get the binary PostgrSQL binary package".

File: docs/Installation.md (hunk 1 above)
Comment (sandoche2k) [reply docs/Installation.md]: I couldn't access the file from 'sudo'

File: docs/Installation.md (hunk 2 above)
Comment (sandoche2k) [reply docs/Installation.md]: Ok

File: docs/Installation.md (hunk 1 above)
Comment (matsduf) [reply docs/Installation.md]: @sandoche2k, do you mean that you cannot do tab completion? If you have the complete path and paste that, then you will be able to edit the file through `sudo`.

File: docs/Installation.md (hunk 2 above)
Comment (mattias-p) [inline docs/Installation.md]: The `useradd` command should probably be invoked with options `-r -c "Zonemaster daemon user"` to make it a system account with a description. Please check on CentOS the user created gets a `nologin` shell, and if not add `-s /usr/sbin/nologin` (possibly adjusting the path to `nologin`).

File: docs/Installation.md [hunk 4]
```diff
@@ -118,19 +118,51 @@ sudo install --mode=755 ./backend_config.ini-postgresql /etc/zonemaster/backend_
...
+```
+
+Configure:
 
 ```sh
+# Switch to root user
+sudo -s
+
+# In the below file modify all instances of "ident" to "md5"
+sudo vi /var/lib/pgsql/9.3/data/pg_hba.conf
```
Comment (mattias-p) [inline docs/Installation.md]: It's good practice to use `sudoedit` instead of `sudo vi`. It limits the write access to the particular file(s) given on the command line.

File: docs/Installation.md [hunk 5]
```diff
@@ -118,19 +118,51 @@ sudo install --mode=755 ./backend_config.ini-postgresql /etc/zonemaster/backend_
...
+# Start PostgreSQL
+sudo systemctl start postgresql-9.3 
+
+# Verify PostgreSQL has started
+sudo systemctl status postgresql-9.3
+
+# Add Zonemaster user
+sudo useradd zonemaster
+
+# Iniitialize Zonemaster database
```
Comment (mattias-p) [inline docs/Installation.md]: Fix spelling

**Comment by sandoche2k** [comment]
@matsduf updated the install instructions

File: docs/Installation.md [hunk 6]
```diff
@@ -118,17 +122,50 @@ sudo install --mode=755 ./backend_config.ini-postgresql /etc/zonemaster/backend_
...
+
+```sh
+sudo /usr/pgsql-9.3/bin/postgresql93-setup initdb
+```
+
+# Configure:
+
+```sh
+# In the below file modify all instances of "ident" to "md5"
+sudo vi /var/lib/pgsql/9.3/data/pg_hba.conf
```
Comment (mattias-p) [inline docs/Installation.md]: It's good practice to use `sudoedit` instead of `sudo vi`. It limits the write access to the particular file(s) given on the command line.

File: docs/Installation.md [hunk 7]
```diff
@@ -65,6 +65,10 @@ sudo cpan -i Zonemaster::Backend
 Check the [declaration of prerequisites] to make sure your preferred combination
 of operating system version and database engine version is supported.
 
+# Add Zonemaster user
+sudo useradd zonemaster
```
Comment (mattias-p) [inline docs/Installation.md]: The `useradd` command should probably be invoked with options `-r -c "Zonemaster daemon user"` to make it a system account with a description. Please check on CentOS the user created gets a `nologin` shell, and if not add `-s /usr/sbin/nologin` (possibly adjusting the path to `nologin`).

File: docs/Installation.md [hunk 8]
```diff
@@ -118,17 +122,50 @@ sudo install --mode=755 ./backend_config.ini-postgresql /etc/zonemaster/backend_
...
+sudo systemctl start postgresql-9.3 
+```
+
+# Verify PostgreSQL has started
+
+```sh
+sudo systemctl status postgresql-9.3
+```
+
+# Iniitialize Zonemaster database
```
Comment (mattias-p) [inline docs/Installation.md]: Fix spelling

File: docs/Installation.md [hunk 9]
```diff
@@ -65,6 +65,10 @@ sudo cpan -i Zonemaster::Backend
 Check the [declaration of prerequisites] to make sure your preferred combination
 of operating system version and database engine version is supported.
 
+# Add Zonemaster user
```
Comment (mattias-p) [inline docs/Installation.md]: Please remove the `#` level 1 heading marker and append a colon instead.

Comment (sandoche2k) [reply docs/Installation.md]: OK

File: docs/Installation.md (hunk 7 above)
Comment (sandoche2k) [reply docs/Installation.md]: Why do we need a loginshell for the zonemaster user here?

File: docs/Installation.md (hunk 6 above)
Comment (sandoche2k) [reply docs/Installation.md]: ok

File: docs/Installation.md [hunk 10]
```diff
@@ -65,6 +65,10 @@ sudo cpan -i Zonemaster::Backend
 Check the [declaration of prerequisites] to make sure your preferred combination
 of operating system version and database engine version is supported.
 
+Add Zonemaster user:
+sudo useradd zonemaster
```
Comment (matsduf) [inline docs/Installation.md]: Replace with the following and move before line 63.

*

Add Zonemaster user that will run Zonemaster Backend:

```
sudo useradd zonemaster
```

File: docs/Installation.md (hunk 7 above)
Comment (mattias-p) [reply docs/Installation.md]: We don't. That's why it should have the `nologin` default shell.

File: docs/Installation.md [hunk 11]
```diff
@@ -118,17 +122,50 @@ sudo install --mode=755 ./backend_config.ini-postgresql /etc/zonemaster/backend_
 sudo install --mode=755 ./zm-centos.sh-postgresql /etc/init.d/zm-centos.sh
 mkdir "$HOME/logs"
 ```
+Get the appropriate PostgreSQL binary package
```
Comment (matsduf) [inline docs/Installation.md]: Add PostgreSQL package repository needed to get the PostgreSQL binary packages.

File: docs/Installation.md (hunk 10 above)
Comment (mattias-p) [inline docs/Installation.md]: Github keeps collapsing my review comments so I'm pasting the conversation here.

@mattias-p said:
> The `useradd` command should probably be invoked with options `-r -c "Zonemaster daemon user"` to make it a system account with a description. Please check on CentOS the user created gets a `nologin` shell, and if not add `-s /usr/sbin/nologin` (possibly adjusting the path to `nologin`).

@sandoche2k replied:
> Why do we need a loginshell for the zonemaster user here?

My point is that we don't. That's why it should have the `nologin` default shell.

File: docs/Installation.md [hunk 12]
```diff
@@ -118,17 +122,50 @@ sudo install --mode=755 ./backend_config.ini-postgresql /etc/zonemaster/backend_
...
-> Can you try to find PostgreSQL package plus perl lib?]
->
+Install the PostgreSQL packages:
 
-Initialize the database:
+```sh
+sudo yum -y install postgresql93 postgresql93-server postgresql93-contrib postgresql93-libs postgresql93-devel perl-DBD-Pg
+```
+
+# To enable PostgreSQL from boot
```
Comment (mattias-p) [inline docs/Installation.md]: Please remove the `#` level one heading mark.

File: docs/Installation.md [hunk 13]
```diff
@@ -118,17 +122,50 @@ sudo install --mode=755 ./backend_config.ini-postgresql /etc/zonemaster/backend_
...
+
+```sh
+sudo systemctl enable postgresql-9.3
+```
+
+# Initialise PostgreSQL
```
Comment (mattias-p) [inline docs/Installation.md]: Please remove the `#` level one heading mark.

File: docs/Installation.md [hunk 14]
```diff
@@ -118,17 +122,50 @@ sudo install --mode=755 ./backend_config.ini-postgresql /etc/zonemaster/backend_
...
+
+```sh
+sudo /usr/pgsql-9.3/bin/postgresql93-setup initdb
+```
+
+# Configure:
```
Comment (mattias-p) [inline docs/Installation.md]: Please remove the `#` level one heading mark.

File: docs/Installation.md [hunk 15]
```diff
@@ -118,17 +122,50 @@ sudo install --mode=755 ./backend_config.ini-postgresql /etc/zonemaster/backend_
...
+
+```sh
+# In the below file modify all instances of "ident" to "md5"
+sudoedit /var/lib/pgsql/9.3/data/pg_hba.conf
+```
+
+# Start PostgreSQL
```
Comment (mattias-p) [inline docs/Installation.md]: Please remove the `#` level one heading mark.

File: docs/Installation.md [hunk 16]
```diff
@@ -118,17 +122,50 @@ sudo install --mode=755 ./backend_config.ini-postgresql /etc/zonemaster/backend_
...
+
+```sh
+sudo systemctl start postgresql-9.3 
+```
+
+# Verify PostgreSQL has started
```
Comment (mattias-p) [inline docs/Installation.md]: Please remove the `#` level one heading mark.

File: docs/Installation.md [hunk 17]
```diff
@@ -118,17 +122,50 @@ sudo install --mode=755 ./backend_config.ini-postgresql /etc/zonemaster/backend_
...
+
+```sh
+sudo systemctl status postgresql-9.3
+```
+
+# Initialize Zonemaster database
```
Comment (mattias-p) [inline docs/Installation.md]: Please remove the `#` level one heading mark.

File: docs/Installation.md [hunk 18]
```diff
@@ -57,6 +57,11 @@ Install Zonemaster::Backend:
 sudo cpan -i Zonemaster::Backend
 ```
 
+Add Zonemaster user:
+```sh
+sudo useradd -r -c "Zonemaster daemon user" zonemaster
+```
```
Comment (matsduf) [inline docs/Installation.md]: Move these four lines after line 66.

File: docs/Installation.md [hunk 19]
```diff
@@ -118,17 +124,51 @@ sudo install --mode=755 ./backend_config.ini-postgresql /etc/zonemaster/backend_
 sudo install --mode=755 ./zm-centos.sh-postgresql /etc/init.d/zm-centos.sh
 mkdir "$HOME/logs"
 ```
+Add PostGreSQL package repository needed to get the appropriate PostgreSQL
```
Comment (matsduf) [inline docs/Installation.md]: Change "PostGreSQL" to "PostgreSQL".
